package/upgrade/upgrade_node.sh

```diff
@@ -29,6 +29,17 @@ cleanup_incomplete_state_file() {
   fi
 }
 
+sparsify_passive_img()
+{
+  # See https://github.com/harvester/harvester/issues/7518
+  echo "Ensuring passive.img is sparse..."
+  mount -o rw,remount ${STATE_DIR}
+  echo "  was: $(du -h ${STATE_DIR}/cOS/passive.img)"
```

Comment:
A similar question, test `${STATE_DIR}/cOS/passive.img` file is exsiting will be even secure.

Reply:
It's extremely unlikely that passive.img will be missing here (someone would have had to intentionally delete it, as in normal operation it's only touched by elemental during `elemental install` or `elemental upgrade`). Still, unlike in the installer where passive.img is guaranteed to exist when we sparsify it (see https://github.com/harvester/harvester-installer/pull/970#discussion_r1953457151), I suppose it's possible the file might not be there at this point if something really weird has happened to the system in the meantime, so I'll add a check...

Reply:
Thanks for the updating.

BTW, besides the file is removed, it is also possible that elemental happens to change the name or file path some day.

The shell script runs into `panic mode` by default if any error happens. Those additional checks cover the exteme cases.